src/canister_id.rs

```diff
@@ -0,0 +1,246 @@
+use hyper::{header::HOST, Request, Uri};
+use ic_agent::export::Principal;
+
+use crate::config::dns_canister_config::DnsCanisterConfig;
+
+/// A resolver for `Principal`s from a `Uri`.
+trait UriCanisterIdResolver: Sync + Send {
```

Comment:
Dumb thought but given this is in a `canister_id` module, does it make sense to just call the trait `Resolve`? i.e
```rs
trait Resolve {
  resolve(&self, uri: &Uri) -> Option<Principal>;
}
```

Reply:
I mean, probably.